pkgs/games/MarbleMarcher/default.nix

```diff
@@ -0,0 +1,97 @@
+{ lib, stdenv, fetchFromGitHub, runtimeShell, cmake, eigen, sfml, libglvnd }:
+
+let
+  pname = "MarbleMarcher";
+
+  # The game doesn’t have any versions in GitHub.
+  # Just taking a commit from the top of the “master” branch.
+  version = "git-2019-04-22";
+
+  src = fetchFromGitHub {
+    owner = "HackerPoet";
+    repo = pname;
+    rev = "478afb8ea9c894d6d8b961583aa557e2f63d6198";
+    sha256 = "1kpn9wnf14wm7867dp3lhb10065h4vzy101gab18zkphlwl1dbwq";
+  };
+
+  esc = lib.escapeShellArg;
+
+  assets = stdenv.mkDerivation {
+    pname = "${pname}-assets-pack";
+    inherit version src;
+
+    phases = [ "installPhase" ];
+
+    installPhase = ''
+      set -eu || exit
+      mkdir -p -- "$out"/assets
+      cp -- "$src"/assets/* "$out"/assets
+    '';
+
+    meta = with lib; {
+      homepage = "https://codeparade.itch.io/marblemarcher";
+      description = "Assets for MarbleMarcher videogame";
+      maintainers = with maintainers; [ unclechu ];
+      platforms = platforms.linux;
+
+      # This assets pack contains:
+      # - A couple of fonts (CC-BY-SA 3.0 and SIL OFL 1.1, with license text files)
+      # - Vertex and fragment shaders under GNU/GPLv2 and later
+      # - Images (png, ico) with not clear license
+      # - Audio files (wav, ogg) with not clear license
+      license = licenses.unfree;
+    };
+  };
+
+  unwrappedName = "${pname}-unwrapped";
```

Comment:
This can be replaced with makeWrapper probably.

Reply:
I’ve tried it first but it failed with a weird error. Let me see if I can get that error message.

Reply:
I removed `phases` completely (as suggested above) and added `postInstall` section (I’ve seen this phase added in other modules).

``` nix
  postInstall = ''
    set -eu || exit
    wrapProgram "$out/bin/MarbleMarcher" \
      --run ${lib.escapeShellArg "cd -- ${lib.escapeShellArg assets}"}
  '';
```

It seems it isn’t evaluated at all.

``` console
$ nix-build -E 'with import ./. { config.allowUnfree = true; }; MarbleMarcher' -o result-MM
/nix/store/0p1sg3gcibdn3i49d0qlwg6818i8ri42-MarbleMarcher-git-2019-04-22
$ file result-MM/bin/MarbleMarcher
result-MM/bin/MarbleMarcher: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), dynamically linked, interpreter /nix/store/mij848h2x5wiqkwhg027byvmf9x3gx7y-glibc-2.33-50/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, not stripped
```

It’s the original binary whilst it supposed to be a text file with few lines of shell code. If I move

``` sh
    wrapProgram "$out/bin/MarbleMarcher" \
      --run ${lib.escapeShellArg "cd -- ${lib.escapeShellArg assets}"}
```

At the end of `installPhase` it fails with:

> /nix/store/86jcaqj8fsf45b0py66fcdi3rfyz47sg-hook/nix-support/setup-hook: line 109: flagsBefore: unbound variable

But I think I see why now. `wrapProgram` isn’t designed with `set -u` in mind and doesn’t check for the name typos.

Reply:
Now replaced with `wrapProgram`